### PR TITLE
CLN: remove compat.lrange

### PR DESCRIPTION
**Pull request by topper-123**
- [x] xref #25725,
- [x] xref #26281, #26289 & #26319

Fourth and final part of the removal of ``compat.lrange`` from the code base.

### CODE REVIEW

File: pandas/tests/internals/test_internals.py [hunk 1]
```diff
@@ -925,13 +924,13 @@ def assert_take_ok(mgr, axis, indexer):
         for mgr in self.MANAGERS:
             for ax in range(mgr.ndim):
                 # take/fancy indexer
-                assert_take_ok(mgr, ax, [])
-                assert_take_ok(mgr, ax, [0, 0, 0])
-                assert_take_ok(mgr, ax, lrange(mgr.shape[ax]))
+                assert_take_ok(mgr, ax, indexer=[])
```
Comment (WillAyd) [inline pandas/tests/internals/test_internals.py]: What was the reason for adding the keyword here?

**Review by WillAyd — COMMENTED** [review]
minor question but not necessarily a block. I think this looks good - nice work on all of this @topper-123

File: pandas/tests/internals/test_internals.py (hunk 1 above)
Comment (topper-123) [reply pandas/tests/internals/test_internals.py]: That was just to aid readability, I found the meaning of those lists became easier to understand. Though I see now it hasnt been done in all the function calls.

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**